src/translate.zig

```diff
@@ -113,20 +113,46 @@ pub fn buffer_from_object(
         return throw(env, key ++ " must be defined");
     }
 
+    return buffer_from_value(env, property, key);
+}
+
+pub fn buffer_from_value(
+    env: c.napi_env,
+    value: c.napi_value,
+    comptime key: [*:0]const u8,
+) ![] u8 {
     var is_buffer: bool = undefined;
-    assert(c.napi_is_buffer(env, property, &is_buffer) == .napi_ok);
+    assert(c.napi_is_buffer(env, value, &is_buffer) == .napi_ok);
 
     if (!is_buffer) return throw(env, key ++ " must be a buffer");
 
     var data: ?*c_void = null;
     var data_length: usize = undefined;
-    assert(c.napi_get_buffer_info(env, property, &data, &data_length) == .napi_ok);
+    assert(c.napi_get_buffer_info(env, value, &data, &data_length) == .napi_ok);
 
     if (data_length < 1) return throw(env, key ++ " must not be empty");
 
     return @ptrCast([*]u8, data.?)[0..data_length];
 }
 
+pub fn copy_buffer_of_length_from_object(env: c.napi_env, object: c.napi_value, comptime length: u8, comptime key: [:0]const u8) ![length]u8 {
+    var property: c.napi_value = undefined;
+    if (c.napi_get_named_property(env, object, key, &property) != .napi_ok) {
+        return throw(env, key ++ " must be defined");
+    }
+
+    const data = try buffer_from_value(env, property, key);
+    if (data.len != length) {
+        return throw(env, key ++ " has incorrect length.");
+    }
+
+    // copy this out of V8 as the underlying data lifetime is not guaranteed.
+    var result: [length]u8 = undefined;
+    std.mem.copy(u8, result[0..], data[0..]);
```

Comment:
Nice comptime! This is also much better than the `napi_create_buffer_copy` that I suggested, which wouldn't actually have been too great here, since it would have allocated another JS Buffer.